Lib/test/test_capi/test_opt.py

```diff
@@ -1370,6 +1370,23 @@ def testfunc(n):
         # Removed guard
         self.assertNotIn("_CHECK_FUNCTION_EXACT_ARGS", uops)
 
+    def test_method_guards_removed_or_reduced(self):
+
+        def testfunc(n):
+            for i in range(n):
+                test_bound_method(i)
+
+
+        testfunc(TIER2_THRESHOLD)
+
+        ex = get_first_executor(testfunc)
```

Comment:
Can you rewrite this to be more like the newer tests, that actually assert the result is correct? Something like this:

```py
        def testfunc(n):
            x = 0
            for i in range(n):
                x += test_bound_method(i)

        res, ex = self._run_with_optimizer(testfunc, TIER2_THRESHOLD)
        self.assertEqual(res, sum(range(TIER2_THRESHOLD)))
```

Reply:
For sure.